apps/docs/content/guides/auth/auth-hooks.mdx

```diff
@@ -118,15 +170,177 @@ Here's an example:
+
+</TabPanel>
+
+<TabPanel id="https-hook-errors" label="HTTP">
+Hooks return status codes based on the nature of the response. These status codes help determine the next steps in the processing flow:
+
+| HTTP Status Code | Description                                                   | Example Usage                                  |
+| ---------------- | ------------------------------------------------------------- | ---------------------------------------------- |
+| 200, 202, 204    | Valid response, proceed                                       | Successful processing of the request           |
+| 429, 503         | Retry-able errors with Retry-after header supported           | Temporary server overload or maintenance       |
+| 403, 400         | Treated as Internal Server Errors and return a 500 Error Code | Malformed requests or insufficient permissions |
+
+Errors are responses which contain status codes 400 and above. On a retry-able error, such as an error with a `429` or `503` status code, HTTP Hooks will attempt up to three retries with a back-off of two seconds.
+
+Here’s a sample HTTP retry schedule:
+
+| Time Since Start (HH:MM:SS) | Event                   | Notes                                             |
+| --------------------------- | ----------------------- | ------------------------------------------------- |
+| 00:00:00                    | Initial Attempt         | Initial invocation begins.                        |
+| 00:00:05                    | Initial Attempt Timeout | Initial invocation must complete.                 |
+| 00:00:07                    | Retry Start #1          | After 2 sec delay, first retry begins.            |
+| 00:00:12                    | Retry Timeout #1        | First retry timeout.                              |
+| 00:00:14                    | Retry Start #2          | After 2 sec delay, second retry begins.           |
+| 00:00:19                    | Retry Timeout #2        | Second retry timeout. Returns an error on failure |
+
+Return a retry-able error by attaching a appropriate status code (`429`, `503` ) and a non-empty `retry-after` header
+
+```jsx
+return new Response(
+  JSON.stringify({
+    error: `Failed to process the request: ${error}`,
+  }),
+  { status: 429, headers: { 'Content-Type': 'application/json', 'retry-after': 'true' } }
+)
+```
+
+Note that all responses, including errors responses, need a `Content-Type` of `application/json` - not specifying the appropriate `Content-Type` will result in the function returning an error response. Supabase Auth will in turn return an Internal Server Error.
+
+</TabPanel>
+</Tabs>
+
+Outside of runtime errors, both HTTP Hooks and Postgres Hooks return timeout errors. Postgres Hooks have **two** seconds to complete processing while HTTP Hooks should complete in **five** seconds. Both HTTP Hooks and Postgres Hooks are run in a transaction do limit the duration of execution to avoid delays in authentication process.
```

Comment:
Perhaps consider dropping these timeouts in the [shared config file](https://github.com/supabase/supabase/blob/master/packages/shared-data/config.ts) for easy updating? Then you can access it like so:

```
<SharedData data="config">auth.[WHATEVER THE REST OF YOUR PATH IS]</SharedData>
```